server/src/main/java/org/opensearch/bootstrap/BootstrapChecks.java

```diff
@@ -222,11 +223,32 @@ static List<BootstrapCheck> checks() {
         checks.add(new OnErrorCheck());
         checks.add(new OnOutOfMemoryErrorCheck());
         checks.add(new EarlyAccessCheck());
+        checks.add(new JavaVersionCheck());
         checks.add(new AllPermissionCheck());
         checks.add(new DiscoveryConfiguredCheck());
         return Collections.unmodifiableList(checks);
     }
 
+    static class JavaVersionCheck implements BootstrapCheck {
```

Comment:
Should this also implement `alwaysEnforce` so that it [cannot by bypassed](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/bootstrap/BootstrapChecks.java#L160)?

Reply:
I am not sure of it: by and large, it is not critical issue which will incur data loss or something, I don't see any checks (besides OutOfMemory) which mandate `alwaysEnforce`.